ibm/service/cis/data_source_ibm_cis_origin_auth.go

```diff
@@ -0,0 +1,187 @@
+			},
+			CisOriginAuthHostName: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS host name",
+				Optional:    true,
+				Default:     "no_host",
+			},
+			CisOriginAuthRequestType: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS Request Type",
+				Optional:    true,
+				Default:     "zone_level",
+			},
+			"origin_pull_settings_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "CIS origin auth settings enabled or disabled",
+			},
+			"origin_pull_certs_list": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Certficate list",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate id",
+						},
+
+						"certificate": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate detail",
+						},
+						"cert_issuer": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate issue",
+						},
+						"cert_signature": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate signature",
+						},
+						"cert_status": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate active or not",
+						},
+						"cert_expires_on": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate expiry time",
+						},
+						"cert_uploaded_on": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certifaate upldate time",
+						},
+						"cert_serial_number": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certifaate upldate time",
+						},
+					},
+				},
+			},
+		},
+	}
+
+}
+
+func dataIBMCISOriginAuthRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).CisOrigAuthSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, _, _ := flex.ConvertTfToCisThreeVar(d.Get(cisDomainID).(string))
+	sess.Crn = core.StringPtr(crn)
+	sess.ZoneIdentifier = core.StringPtr(zoneID)
+
+	request_type := d.Get(CisOriginAuthRequestType).(string)
+
+	if request_type == "zone_level" {
+
+		// Get Zone Origin Pull Settings
+		zoneSettingsOpt := sess.NewGetZoneOriginPullSettingsOptions()
+		zoneSettingsResult, zoneSettingsResponse, zoneSettingsErr := sess.GetZoneOriginPullSettings(zoneSettingsOpt)
+
+		if zoneSettingsErr != nil || zoneSettingsResponse == nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error Getting Zone Level Origin Pull Settings: %s", zoneSettingsErr))
```

Comment:
Retrun the "zoneSettingsResponse" also along with the error in all return statements